### PR TITLE
fix: slider handle negative data-start values

### LINKED ISSUE #11390: Slider handler, wrapped as react component, value updates with offset by 1 when clicking on negative step
Hello!
I hope this issue finds you joyful :) 

## What should happen?

I'm expecting my reactified slider handle to move where the user has clicked, all the times.

## What happens instead?

slider

The slider handle is correctly setting it's value when clicking on positive numbers. But when clicking on negative the `e.target.firstChild.attributes.getNamedItem('aria-valuenow').value` from `$('.slider').on('changed.zf.slider)'` event is having an offset by 1 :/

## Possible Solution

I wanted to use some method to update the handler value again when consuming the event $('.slider').on('changed.zf.slider). Something like https://github.com/zurb/foundation-sites/blob/develop/js/foundation.slider.js#L341 but this is not working and it seems like a workaround rather than a fix.

```
    $('.slider').on('changed.zf.slider', e => {
      let sliderVal = parseInt(
        e.target.firstChild.attributes.getNamedItem('aria-valuenow').value,
        10,
      )
      if (sliderVal < 0) {
        sliderVal -= 1
        $('.slider').foundation('_setValues', jQuery('.slider'), sliderVal)
      }
      this.props.changeMoodMeterFormField('pleasantness', sliderVal)
    })
```

Any ideas what may be causing this problem?

## Test Case and/or Steps to Reproduce

Test Case:

I've tried to recreate the issue in codepen https://codepen.io/altras/pen/oyKPJG?editors=0010 It's using our css for styling but even without it it behaves the same.

## Context

We want to include the slider in a journey
clicking-numbers to map users pleasantness and energy. But when using mouse clicking behavior the slider is not updating properly if we want to select a negative step.

## Your Environment

- Foundation version(s) used: `6.4.3 foundation.min.js`
- Browser(s) name and version(s):
  - Chrome Version 67.0.3396.99 (Official Build) (64-bit)
  - Firefox Quantum 61.0.1 (64-bit)
- Operating System and version (desktop or mobile): MacOS 10.13.4 (17E202)

## Checklist (all required):

- [x] I have read and follow the CONTRIBUTING.md document.
- [x] There are no other issues similar to this one.
- [x] The issue title is descriptive.
- [x] The template is correctly filled.

### PR DESCRIPTION
**Pull request by veselinn**
## Description
The Slider handles negative data-start values. The change is isolated only in `_adjustValue`.

- Closes https://github.com/zurb/foundation-sites/issues/11390

## Motivation and Context
Having a negative data-start value extends the use cases for the Slider.

## Types of changes

- [ ] Documentation
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing
      functionality to change)

## Checklist (all required):

- [x] I have read and follow the [CONTRIBUTING](CONTRIBUTING.md) document.
- [x] There are no other pull request similar to this one.
- [x] The pull request title is descriptive.
- [x] The template is fully and correctly filled.
- [x] The pull request targets the right branch (`develop` or `develop-v...`).
- [x] My commits are correctly titled and contain all relevant information.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly to my changes (if relevant).
- [x] I have added tests to cover my changes (if relevant).
- [x] All new and existing tests passed.

### CODE REVIEW

File: js/foundation.slider.js [hunk 1]
```diff
@@ -426,7 +426,11 @@ class Slider extends Plugin {
     else {
       val = value;
     }
-    left = val % step;
+    if (val >= 0) {
+      left = val % step;
+    } else if (val < 0) {
```
Comment (DanielRuf) [inline js/foundation.slider.js]: Shouldn't it be `else`?

**Comment by veselinn** [comment]
@DanielRuf Removed the unnecessary `if`. Shame on me for not noticing it. Thank you for the comment ^_^.

Should I squash the commits in this PR ? :thinking:

**Review by ncoden — APPROVED** [review]
LGTM. Slider needs some cleaning btw.

**Comment by ncoden** [comment]
@veselinn Could you add some unit tests for negative ranges ?

**Comment by veselinn** [comment]
@ncoden Added some unit tests for `_adjustValue` with https://github.com/zurb/foundation-sites/pull/11409/commits/9851a17ffeb335cb27fe157980160e595276cf7f .

Dunno if you had such unit test in mind, but this was the laziest solution I could think of. Please tell me if more is needed for this PR. Thanks for taking a look.

**Review by DanielRuf — APPROVED** [review]
